src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1046,7 +1046,8 @@ void MarkScope (IMetadataScope scope)
 		protected virtual void MarkSerializable (TypeDefinition type)
 		{
 			MarkDefaultConstructor (type);
-			MarkMethodsIf (type.Methods, IsSpecialSerializationConstructor);
+			if (!_context.IsFeatureExcluded ("deserialization"))
```

Comment:
I think it'd be better to do the check at caller of `MarkSerializable`

Reply:
The default ctor needs to be retained.

Reply:
Why?

Reply:
I get some runtime assertions without it, I think things just worked by accident because some types whose ctors are needed at runtime were marked as serializable.

Reply:
Also, not sure what the default ctor has to do with serialization at all.

Reply:
Yeah, this looks like quirk for `XmlSerializer` support which has constructor of System.Type and requires default constructor to be present but no SerializableAttribute to be applied!

Made a new tracking issue #957